libs/common/src/main/java/org/opensearch/common/crypto/KeyStoreFactory.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.crypto;
+
+import org.bouncycastle.crypto.CryptoServicesRegistrar;
+import org.opensearch.common.SuppressForbidden;
+
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchProviderException;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static org.opensearch.common.crypto.KeyStoreType.SECURE_KEYSTORE_TYPES;
+import static org.opensearch.common.crypto.KeyStoreType.inferStoreType;
+
+/**
+ * Restricts types of keystores to PKCS#11 and BCFKS when running in FIPS JVM.
+ * Returns the keystore from specified provider or otherwise follows the priority of
+ * declared security providers and their support for different keystores.
+ */
+public final class KeyStoreFactory {
```

Comment:
Does this need to be in `common`? On the surface it seems like `ssl-config` might be a better place?

Reply:
See my comment above: https://github.com/opensearch-project/OpenSearch/pull/14912#discussion_r1907900335

> If I understand correctly, this change is needed because KeyStoreFactory is referenced from [KeyStoreWrapper](https://github.com/opensearch-project/OpenSearch/pull/14912/files#diff-9e818301234eedaabb31754070e78a276a0a8bf5a9e76dc3d2e6088fc3a59af9R461) which is in the server module. If the KeyStoreFactory class is moved to opensearch-ssl-config, then would it also be necessary to add a new dependency to the server module on api project(":libs:opensearch-ssl-config")?

We can put it in the `opensearch-ssl-config` lib and then add a dependency on that lib in the server module.